tests/MongoDB.Driver.Core.TestHelpers/CoreExceptionHelper.cs

```diff
@@ -26,64 +26,90 @@ namespace MongoDB.Driver.Core.TestHelpers
                         var innerException = new Exception();
                         return new MongoConnectionException(connectionId, message, innerException);
-                    }
+                }
 
                 case "MongoConnectionClosedException":
-                    {
+                {
                         var clusterId = new ClusterId(1);
                         var serverId = new ServerId(clusterId, new DnsEndPoint("localhost", 27017));
                         var connectionId = new ConnectionId(serverId, 1);
                         return new MongoConnectionClosedException(connectionId);
-                    }
+                }
 
                 case "MongoCursorNotFoundException":
-                    {
+                {
                         var clusterId = new ClusterId(1);
                         var serverId = new ServerId(clusterId, new DnsEndPoint("localhost", 27017));
                         var connectionId = new ConnectionId(serverId, 1);
                         var cursorId = 1L;
                         var query = new BsonDocument();
                         return new MongoCursorNotFoundException(connectionId, cursorId, query);
-                    }
+                }
 
                 case "MongoNodeIsRecoveringException":
-                    {
+                {
                         var clusterId = new ClusterId(1);
                         var serverId = new ServerId(clusterId, new DnsEndPoint("localhost", 27017));
                         var connectionId = new ConnectionId(serverId, 1);
                         var result = new BsonDocument();
                         return new MongoNodeIsRecoveringException(connectionId, null, result);
-                    }
+                }
 
                 case "MongoNotPrimaryException":
-                    {
+                {
                         var clusterId = new ClusterId(1);
                         var serverId = new ServerId(clusterId, new DnsEndPoint("localhost", 27017));
                         var connectionId = new ConnectionId(serverId, 1);
                         var result = new BsonDocument();
                         return new MongoNotPrimaryException(connectionId, null, result);
-                    }
+                }
+
+                // custom errors next:
+                case "IOExceptionWithNetworkUnreachableException":
+                {
+                    var innerException = new SocketException(errorCode: (int)SocketError.NetworkUnreachable);
+                    return new IOException("IoExceptionWithNetworkUnreachableException", innerException);
+                }
+
+                case "IOExceptionWithTimedOutException":
+                {
+                    var innerException = new SocketException(errorCode: (int)SocketError.TimedOut);
+                    return new IOException("IOExceptionWithTimedOutException", innerException);
```

Comment:
Consider a recursive call to reduce code duplication:

    var innerException = CreateException("TimedoutSocketException");
    return new IOException("IOExceptionWithTimedOutException", innerException);